.github/workflows/ci.yml

```diff
@@ -104,6 +104,23 @@ jobs:
           doxygen docs/doxygen/config.doxyfile 2>&1 | tee doxyoutput.txt
           if [[ "$(wc -c < doxyoutput.txt | bc)" = "0" ]]; then exit 0; else exit 1; fi
 
+  build-checks:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Update submodules
+        run: git submodule update --init --checkout
```

Comment:
Should this have the --recursive flag in case there are nested submodules?

Reply:
This test just uses the FreeRTOS-Kernel sources which does not have any submodules itself.
It would be unnecessary to copy files other than the FreeRTOS-Kernel.

I would not mind adding that flag though. What is your opinion @yanjos-dev?

Thanks

Reply:
If the flag isn't required for now then we can go ahead and not use it for now.